### PR TITLE
8307806: Rename Atomic::fetch_and_add and friends

### PR DESCRIPTION
**Pull request by kimbarrett**
Please review this renaming of Atomic::fetch_and_add and friends to be
consistent with the naming convention recently chosen for atomic bitops.  That
is, make the following name changes for class Atomic and it's implementation:

- fetch_and_add => fetch_then_add
- add_and_fetch => add_then_fetch

Testing:
mach5 tier1-3
GHA testing

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8307806](https://bugs.openjdk.org/browse/JDK-8307806): Rename Atomic::fetch_and_add and friends

### Reviewers
 * [Stefan Karlsson](https://openjdk.org/census#stefank) (@stefank - **Reviewer**)
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**) ⚠️ Review applies to [f527511b](https://git.openjdk.org/jdk/pull/13896/files/f527511b650c4ec1b2102b0b5e9d84eff458c9fb)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13896/head:pull/13896` \
`$ git checkout pull/13896`

Update a local copy of the PR: \
`$ git checkout pull/13896` \
`$ git pull https://git.openjdk.org/jdk.git pull/13896/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13896`

View PR using the GUI difftool: \
`$ git pr show -t 13896`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13896.diff">https://git.openjdk.org/jdk/pull/13896.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13896#issuecomment-1542187621)

### CODE REVIEW

**Review by stefank — APPROVED** [review]

**Review by dholmes-ora — APPROVED** [review]
LGTM!

Thanks

**Comment by kimbarrett** [comment]
I've merged after the genzgc integration (there were a couple of simple conflicts, where I took the new code but
applied the renaming to it).  I then searched for and renamed some "new" occurrences of fetch_and_add (some
were in gc/x).  So not a completely trivial merge.  Reran mach5 tier1-3.
@stefank and @dholmes-ora - still okay?

**Review by stefank — APPROVED** [review]
Looks good. I don't think you need to wait for David's approval of the ZGC changes.

**Comment by kimbarrett** [comment]
Thanks @stefank and @dholmes-ora for reviews.

File: src/hotspot/share/gc/z/zRelocationSet.cpp
Comment (dholmes-ora) [inline src/hotspot/share/gc/z/zRelocationSet.cpp]: Why was this removed?

Comment (dholmes-ora) [reply src/hotspot/share/gc/z/zRelocationSet.cpp]: Never mind this seems to be a quirk of the github UI.

**Pull request closed without merging**
